core/unit_test/TestArray.cpp

```diff
@@ -187,6 +187,102 @@ constexpr bool test_to_array() {
 
 static_assert(test_to_array());
 
+// making sure we cover both const and non-const cases by having a function that
+// writes to an array and another one that reads from it
+// also checking that it supports host device annotations
+template <class T, size_t N>
+KOKKOS_FUNCTION constexpr void iota(Kokkos::Array<T, N>& a, T value) {
+  for (auto& e : a) {
+    e = value++;
+  }
+}
+
+template <class T, size_t N>
+KOKKOS_FUNCTION constexpr T accumulate(Kokkos::Array<T, N> const& a, T init) {
```

Comment:
For a reference implementation that uses the `init` directly in the code (no need for `acc`): https://github.com/llvm/llvm-project/blob/c28b1a19aadff97b369889aee084073a181cfda8/libcxx/include/__numeric/accumulate.h#L27. Not requesting you to change anything if you're not willing to :smile: